airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -144,13 +144,12 @@ def get_conn(self) -> DataFactoryManagementClient:
         tenant = conn.extra_dejson.get('extra__azure_data_factory__tenantId')
         subscription_id = conn.extra_dejson.get('extra__azure_data_factory__subscriptionId')
 
-        credential = None
         if conn.login is not None and conn.password is not None:
             credential = ClientSecretCredential(
-                client_id=conn.login, client_secret=conn.password, tenant_id=tenant
+                client_id=conn.login, client_secret=conn.password, tenant_id=tenant  # type: ignore
             )
         else:
-            credential = DefaultAzureCredential()
+            credential = DefaultAzureCredential()  # type: ignore
```

Comment:
Add `credential: Any` before the `if` instead. These credential classes are opaque and internal to Azure, and we are passing them directly back to Azure. This is better than `# type: ignore` because it is more obvious what kind of errors we are trying to ignore.